File: src/web/schemas/intel_report.py
"""Intel Report Pydantic schemas for API serialization.

Part of v25.0 Intel-Portfolio Analysis (FR-IA003/004).
"""

from __future__ import annotations

from pydantic import BaseModel, Field


class ReportFactor(BaseModel):
    """A single analysis factor contributing to the signal."""

    category: str = ""
    impact: str = ""
    weight: float = 0.0
    description: str = ""


class ReportPositionContext(BaseModel):
    """Position-aware context within a report."""

    cost_price: float | None = None
    shares: int | None = None
    pnl_percent: float | None = None
    advice: str = ""
    key_levels: dict[str, float | None] | None = None


class IntelReportResponse(BaseModel):
    """Single intel report returned by the API."""

    id: str
    symbol: str
    stock_name: str = ""
    intel_item_ids: list[str] = Field(default_factory=list)
    refresh_cycle: str = ""
    action: str = "hold"
    signal: str = "neutral"
    confidence: float = 0.5
    summary: str = ""
    factors: list[ReportFactor] = Field(default_factory=list)
    position_context: ReportPositionContext | None = None
    risk_warnings: list[str] = Field(default_factory=list)
    outlook: str = ""
    reasoning: list[str] = Field(default_factory=list)
    intel_summary: str = ""
    model_used: str = ""
    generated_at: str = ""
    created_at: str = ""
    thread_id: str | None = None
    is_read: bool = False


class ReportListResponse(BaseModel):
    """Paginated list of reports."""

    reports: list[IntelReportResponse] = Field(default_factory=list)
    total: int = 0


class ReportUnreadCountResponse(BaseModel):
    """Unread report count."""

    count: int = 0


class ChatFromReportResponse(BaseModel):
    """Response after creating a chat thread from a report."""

    thread_id: str
    initial_message: str = ""

File: frontend/src/types/intel-report.ts
/** Intel Report types for v25.0 Intel-Portfolio Analysis. */

export interface ReportFactor {
  category: string
  impact: string
  weight: number
  description: string
}

export interface ReportPositionContext {
  cost_price: number | null
  shares: number | null
  pnl_percent: number | null
  advice: string
  key_levels: { support: number; resistance: number } | null
}

export interface AffectedSectors {
  bullish?: Array<{ sector: string; logic: string }>
  bearish?: Array<{ sector: string; logic: string }>
  neutral?: Array<{ sector: string; logic: string }>
}

export interface IntelReport {
  id: string
  symbol: string
  stock_name: string
  intel_item_ids: string[]
  refresh_cycle: string
  action: "buy" | "sell" | "hold" | "watch"
  signal: "bullish" | "bearish" | "neutral"
  confidence: number
  summary: string
  factors: ReportFactor[]
  position_context: ReportPositionContext | null
  risk_warnings: string[]
  outlook: string
  reasoning: string[]
  intel_summary: string
  model_used: string
  generated_at: string
  created_at: string
  thread_id: string | null
  is_read: boolean
  is_macro: boolean
  affected_sectors?: AffectedSectors
}

export interface ReportListResponse {
  reports: IntelReport[]
  total: number
}

export interface ReportListParams {
  symbol?: string
  unread_only?: boolean
  limit?: number
  offset?: number
  macro_only?: boolean
}
